cmd/snap-confine/snap-confine.c

```diff
@@ -648,25 +677,31 @@ static void enter_non_classic_execution_environment(sc_invocation *inv,
 	// Init and check rootfs_dir, apply any fallback behaviors.
 	sc_check_rootfs_dir(inv);
 
+	// Set up a device cgroup, unless the snap has been allowed to manage the
+	// device cgroup by itself.
+	if (!is_device_cgroup_self_managed(inv)) {
```

Comment:
This is checked after grabbing the per-snap lock file. Having said that it is also accepting the absence of the file as valid state. I think I agree with @pedronis' remark about only looking after one of the parts that wait for system-key to be set.